src/vs/workbench/services/languageRuntime/common/languageRuntimeDataExplorerClient.ts

```diff
@@ -33,6 +32,79 @@ export enum DataExplorerClientStatus {
 	Error
 }
 
+export interface DataExplorerUiEvent {
+	/**
+	 * Unique resource identifier for routing method calls.
+	 */
+	uri: string;
+
+	/**
+	 * Method name, as defined
+	 */
+	method: DataExplorerFrontendEvent;
+
+	/**
+	 * Data for event
+	 */
+	params: ReturnColumnProfilesEvent | DataUpdateEvent | SchemaUpdateEvent;
+}
+
+/**
+ * An instance of a data explorer backend implementation, whether provided by
+ * a language runtime, in-application embedded database (i.e. DuckDB), remote
+ * connection, etc.
+ */
+export interface IDataExplorerBackendClient extends Disposable {
+	clientId: string;
+	onDidClose: Event<void>;
+	onDidSchemaUpdate: Event<SchemaUpdateEvent>;
+	onDidDataUpdate: Event<DataUpdateEvent>;
+	onDidReturnColumnProfiles: Event<ReturnColumnProfilesEvent>;
+	getState(): Promise<BackendState>;
+	getSchema(columnIndices: Array<number>): Promise<TableSchema>;
+	getDataValues(columns: Array<ColumnSelection>, formatOptions: FormatOptions): Promise<TableData>;
+	getRowLabels(selection: ArraySelection, formatOptions: FormatOptions): Promise<TableRowLabels>;
+	exportDataSelection(selection: TableSelection, format: ExportFormat): Promise<ExportedData>;
+	setColumnFilters(filters: Array<ColumnFilter>): Promise<void>;
+	setRowFilters(filters: Array<RowFilter>): Promise<FilterResult>;
+	setSortColumns(sortKeys: Array<ColumnSortKey>): Promise<void>;
+	getColumnProfiles(callbackId: string, profiles: Array<ColumnProfileRequest>, formatOptions: FormatOptions): Promise<void>;
+}
```

Comment:
Maybe there is a better place to put these changes

Reply:
Seems fine for now. I think it'd be good to expose these types elsewhere but that'd require more of a refactor than I think we want for this PR.